tests/test_model_evaluation.py

```diff
@@ -72,3 +85,305 @@ def test_eval_function_calls(eval_func):
 
     evaluators = eval_func(estimator, X, y, evaluators)
     assert evaluators[0].aggregate_call  # type: ignore
+
+
+class _MockRandomEvaluator(Evaluator):
+    """Produce a random evaluation."""
+
+    def __init__(self, sample_from_rng_with_distribution):
+        """Produce a random evaluation, drawn using the given sampling function."""
+        self._random_state = None
+        self._results = []
+        self._sample_from_rng_with_distribution = sample_from_rng_with_distribution
+        pass
+
+    def prepare(self, estimator, X, y=None, random_state=None):
+        """Initialise using data; in this case, only the random state is used."""
+        self._random_state = random_state
+
+    def _evaluate(self, estimator, X, y):
+        """Evaluate estimator; in this case, the evaluation is random."""
+        # pass through/seed/create new rng as appropriate
+        self._random_state = check_random_state(self._random_state)
+        result = self._sample_from_rng_with_distribution(self._random_state)
+        # intended semantics is that repeated calls *append* to internal state
+        self._results.append(result)
+
+        return result
+
+    evaluate_all = _evaluate
+    evaluate_train = _evaluate
+    evaluate_test = _evaluate
+
+    def get_results(self):
+        """Return (random) evaluation."""
+        return self._results
+
+
+model_evaluators = [crossval_model, bootstrap_model]
+random_seeds = [42, np.random.RandomState()]
+
+
+@pytest.mark.parametrize(
+    "eval_func, random_state", itertools.product(model_evaluators, random_seeds)
```

Comment:
I don't think you actually need `product` here - I believe the default behaviour of `parametrize` is to do a Cartesian product of the arguments. See how here:

https://github.com/gradientinstitute/aboleth/blob/master/tests/test_kernels.py#L28